File: rec-core/src/main/java/net/kimleo/rec/rule/RuleRunner.java
package net.kimleo.rec.rule;

import net.kimleo.rec.Pair;
import net.kimleo.rec.repository.RecRepository;
import net.kimleo.rec.repository.selector.Selector;

import java.util.List;

public class RuleRunner {

    private final RecRule rule;
    private final Selector selector;

    public RuleRunner(RecRule rule, Selector selector) {
        this.rule = rule;
        this.selector = selector;
    }

    public Pair<Boolean, List<Result>> runOn(RecRepository repo) {
        return rule.verify(selector.findAll(repo));
    }
}

File: src/main/kotlin/net/kimleo/rec/rule/RuleRunner.kt
package net.kimleo.rec.rule

import net.kimleo.rec.repository.RecRepository
import net.kimleo.rec.repository.selector.Selector

class RuleRunner(val rule: RecRule, val selector: Selector) {
    fun runOn(repo: RecRepository): Pair<Boolean, List<Result>> {
        return rule.verify(selector.findAll(repo))
    }
}